cabal-install/Distribution/Solver/Modular/Dependency.hs

```diff
@@ -212,8 +219,13 @@ data QualifyOptions = QO {
 --
 -- NOTE: It's the _dependencies_ of a package that may or may not be independent
 -- from the package itself. Package flag choices must of course be consistent.
+--
+-- NOTE 2: This should be called on _all_ dependencies of a package. If it gets
+-- called on a subset of the dependencies, we might construct invalid
+-- quantifiers. In particular, we might conclude that a dependency of a test
+-- suite is not shared with the library and hence is independent.
```

Comment:
I think this is inconsistent with #3357.  It added `requalifyDeps`, which can call `qualifyDeps` on only a subset of dependencies: https://github.com/haskell/cabal/blob/37325ca24b37ab2a96b1947b6da301c5d012f168/cabal-install/Distribution/Solver/Modular/Linking.hs#L290